banking_system.rb

```diff
@@ -0,0 +1,133 @@
+class Account
+	@@number = 23456789101
+	@@balance = 0
+	def initialize
+		@acc_number = @@number
+	end
+
+	def open_account
+		@@number += 1
+		@acc_number = @@number
+	end
+
+	def deposit_money(amount)
+		@@balance += amount
+	end
+
+	def withdraw_money(amount)
+		if @@balance <= 0 || @@balance <= amount
+			return false
+		else
+			@@balance -= amount
+			return true
+		end
+	end
+
+	def get_balance
+		return @@balance
+	end
+
+	def interest(rate,time)
+		interst = (@@balance*rate*time)/100
+	end
+
+	def type(x)
+		@@typee = x
+	end
+
+	def give_type
+		return @@typee
+	end
+
+end
+
+class Current_Account < Account
+
+	def interest(rate,time)
+		interst = super
+	end
```

Comment:
Are we using the local variable `interst` anywhere, if not then why are we assigning it?